### ISSUE TITLE
gearswap issue with include self_command(command)

### ISSUE THREAD

**Issue by smd111** [Windower/Lua#883]
sence friday the 30th(one of the updates) these commands no longer work but thay worked with no issues before that update
gs c tstavetouse <-- from my mage staves include
gs c stepcount <-- from my subjob include
the funny thing is that these commands work just fine
gs c tstopsteps <-- from my subjob include
gs c tchangemagestaff <-- from my mage staves include
1. i do not use motes includes but it feels like the time i tried to use them when i first started with gearswap i could never get them to work
2. i do not want to use motes includes for the same reason above
3. i have tried this with all other addons and plugins disabled(with only gearswap loaded) and the issues still ocures

here is my job lua
http://pastebin.com/HMVBR4Wg <-- PLD.lua
here are my includes
http://pastebin.com/j1uY2YS2 <--mage staves include
http://pastebin.com/RLE1wUvV <--weapon skill include
http://pastebin.com/H3quw7j4 <-- sub job include

**Comment by Byrth** [comment]
Do they give errors or just fail to work?

**Comment by smd111** [comment]
thay fail to even get registered by gearswap
i tried this on both of them

```
if command == 'stepcount' then
        add_to_chat(123,'yes')
```

and it never got kicked out

**Comment by smd111** [comment]
update
i just triede to 
1. just use one or the other include by them selvs with no difference
2. i tried to rename the commands to "test" still no change

**Comment by Byrth** [comment]
It seems that the function is not getting called. Can you please check if you define it twice?

**Comment by smd111** [comment]
i dont call function  self_command(command)
in my includes its(respectively)
function sub_jobs_command(command)
function equip_elemental_magic_gear_command(command)
in my job lua its

```
function self_command(command)
    sub_jobs_command(command)
    equip_elemental_magic_gear_command(command)
end
```

i even tried other jobs like blm and run that thay worked on before the update but no good

**Comment by smd111** [comment]
i just tried to put)
    add_to_chat(123,'yes ??') --ware ?? is job ms sj depending on the file its in
just below the function line and nothing even when i call on the ones that work
but i just enabled debugmode and it shows that the command function is being called

like this
function equip_elemental_magic_gear_command(command)
    add_to_chat(123,'yes ms')

**Comment by joshk6656** [comment]
Should your includes be inside get_sets? I'm not 100% familiar with GearSwap but you maybe defining functions multiple times if get_sets is called multiple times?

**Comment by smd111** [comment]
yes the
    include('Weapon Skill include.lua')
    include('Mage Staves include.lua')
    include('Sub Job include.lua')
are in my job.lua function get_sets
and no functions are named the same in any files

**Comment by joshk6656** [comment]
Yes, but if get_sets is called multiple times, the functions will be defined multiple times I believe since you are re-including it. This is a guess on my part. 

Try pulling the includes out of get_sets and defining them at the top and see what happens.

**Comment by smd111** [comment]
i only have 1 get_sets for each job and thats in the job.lua
i learned that months ago when i frist started using gearswap

**Comment by joshk6656** [comment]
That's not what I mean. I mean gearswap maybe calling get_set multiple times (I'm not 100% on this part) which is causing your include files to be included multiple times and making the functions inside your include be defined multiple times (also a guess).

Can you just pull your includes out of get_sets and define them at the top of the file, or do they need to be defined in get_sets?

**Comment by smd111** [comment]
if thats the case then gearswap would be bugged because get_sets is only supost to be call when you change jobs or relode gearswap and never again and none of my code calls get_sets only gs does that

**Comment by joshk6656** [comment]
Could you just see if it works if you move the include out of get_sets?

**Comment by smd111** [comment]
it wont because i have variables that need to be in get_sets for them to work(even the commands)
and the includes need to be called in get_sets for that

if you dont call an include in get_sets all the variables would be local not globa;

**Comment by Byrth** [comment]
It's fine that it's in get_sets, because get_sets is only called once. I meant for you to check whether equip_elemental_magic_gear_command(command) is only defined once.

**Comment by smd111** [comment]
yes only once

just so you know every time i try something new i revert beck to the original code before trying it again

i.e if i add
        add_to_chat(123,'self-command = '.. tostring(command))
before i try to do this
        add_to_chat(123,'yes')
i use notepad++ revert controls to go back to the way it was when i opened notepad++ the last time

**Comment by smd111** [comment]
i just tried 
    print('self-command = '.. tostring(command))
to print to console but no luck

**Comment by Byrth** [comment]
Can you please put a print statement above the equip_elemental_magic function call in self_command?

**Comment by smd111** [comment]
this is what it looks like right now

```
function self_command(command)
    print('self-command = '.. tostring(command))
    sub_jobs_command(command)
    equip_elemental_magic_gear_command(command)
end
```

and its not printing anything

**Comment by smd111** [comment]
i just tried adding set_language('english') in get_sets still no good

**Comment by smd111** [comment]
mote said that he tried my files and there working for him(on bluegartr)

**Comment by Byrth** [comment]
So wait, the sub_jobs_commands work?

**Comment by smd111** [comment]
thay all work for mote but only one of them does for me(in each of the includes)

im going to try the dev version to see if its the live version thats the issue

**Comment by smd111** [comment]
nope it does not work for me in the dev version eather

**Comment by smd111** [comment]
i just deleated all addons,libs,res,gearswap and redownloaded them still no good
i even coppied the posts i made to pastbin as mote did still no good

**Comment by Byrth** [comment]
Do you have multiple copies of your includes, perhaps in different folders?

**Comment by smd111** [comment]
yes but there just backup copies

**Comment by joshk6656** [comment]
Are they named the same or differently? 

If they are named the same I'd try pulling the backups out of the gearswap directory structure entirely and see if works then.

**Comment by smd111** [comment]
i figured it out now i got it fixed
i was being a DUMB@$$ i some how was editing my backups(i dont know how i did that) but on my normal one i had comented out few lines because i was testing something
its been a bad few weeks my sister is in the hospital
its all fixed

**Comment by z16** [comment]
For the record, includes (and everything else) can be screwed up if you define it in `get_sets`, since includes are then loaded after your own definitions and can overwrite them. They should not be in there, but instead in the global scope.

**Comment by smd111** [comment]
what the problem was is i was in the middle of doing something when my sister whent in to hospital and i had to stop and forgot what i was doing because of it and thats what happened

and yes she went in to hospital on 5/30/14

**Comment by z16** [comment]
That was not relating to your problem (although that could have very well explained it).

**Comment by smd111** [comment]
here was the issue
i have(for my includes) 2 files
for weapon skill include i have
Weapon Skill include.lua --for long name
and
WSi.lua -- for short name when an error shows up

in wsi i comented out some things for testing
and for some reason i was editing my backup files(i dont remember for what for) and did not take them out of notepad++ so while i thought i was looking at my normal files i was looking at my backups and did not realize it so i was not editing/looking at my live files but my backups
and when notepad++closes it will rember what files were open the last time it closed

thats why im a dumb@$$

and this comes because my sister whent to hospital

**Comment by Byrth** [comment]
Nice to hear you solved the problem. I hope your sister is better soon.

Feel free to open another issue of you have another problem.
